Fix oversize JSON response

mwbenowitz opened the pull request:
Currently the search API returns a full set of editions for the search result response. This is unecessary as only the top three will ever be displayed. Too large responses both affect performance and can cause axios to throw a 500 error when handling a response of over 5MB (roughly).

This also implements a stub function that will be able to sort/filter editions depending on the current search criteria, which is a task that is currently in the backlog. This implements no functionality, but provides a method and testing framework that can be built into this.

mwbenowitz committed:
Fix oversize JSON response

Currently the search API returns a full set of editions for the search result response. This is unecessary as only the top three will ever be displayed. Too large responses both affect performance and can cause axios to throw a 500 error when handling a response of over 5MB (roughly).

This also implements a stub function that will be able to sort/filter editions depending on the current search criteria, which is a task that is currently in the backlog. This implements no functionality, but provides a method and testing framework that can be built into this.

nachocs commented:
We will need a new property on search work items to know the total of instances, if they are filtered to only 3

mwbenowitz commented:
> We will need a new property on search work items to know the total of instances, if they are filtered to only 3

Good point, would calling it `edition_count` work? Its easy to implement

nachocs commented:
> > We will need a new property on search work items to know the total of instances, if they are filtered to only 3
> 
> Good point, would calling it `edition_count` work? Its easy to implement

Sure, that should be ok. We'll need a FE ticket to adapt this, @seanredmond

mwbenowitz committed:
Add edition_count field to _source response

mwbenowitz committed:
Add error catching to edition_range calculations

There are edge cases where instances are missing or they cannot be counted. This adds several cases to ensure that these are caught.

This also implements edge case checking if the first and last year are the same and ensures that only the single year is output
